Kernel/Arch/aarch64/Registers.h

```diff
@@ -700,11 +700,11 @@ static_assert(sizeof(SCTLR_EL1) == 8);
 // https://developer.arm.com/documentation/ddi0601/2022-09/AArch64-Registers/MIDR-EL1--Main-ID-Register?lang=en
 // MIDR_EL1, Main ID Register
 struct alignas(u64) MIDR_EL1 {
-    int Revision : 4;
-    int PartNum : 12;
-    int Architecture : 4;
-    int Variant : 4;
-    int Implementer : 8;
+    u8 Revision : 4;
```

Comment:
I'm not sure I understand this change. Is making these `unsigned` what helps?

Changing the size of the int type shouldn't have a big effect on the itanium abi if there's a bitfield size, I thought. Do you have a godbolt example where this makes a difference?

Reply:
Removed from here as it belongs to #17201 (short answer: need not to use mask when reading revision—it was definitely related to the sign, as you noted, the snippet Iʼve used did not land yet in a PR, though).